src/persist-client/README.md

```diff
@@ -0,0 +1,87 @@
+# Persist
+
+Persist is an implementation detail of STORAGE. Its "public" API is used only
+by STORAGE and code outside of STORAGE should be talking to STORAGE, not
+persist. However, having this strong API boundary between the two allows for the
+two teams to execute independently.
+
+Persist's primary abstraction is a "shard", which is a durable and definite
+[Time-Varying Collection (TVC)]. Persist requires that the collection's "data"
+is key-value structured (a `()` unit value is fine) but otherwise allows the
+key, value, time, and diff to be abstract in terms of the [Codec] and [Codec64]
+encode/decode traits. As a result, persist is independent of Materialize's
+internal data formats: `Row` etc.
+
+[Time-Varying Collection (TVC)]: https://github.com/MaterializeInc/materialize/blob/13603a6ee1c13eaa243948fc6a99e8fe2a9d0cdd/doc/developer/platform/formalism.md#time-varying-collections
+[Codec]: mz_persist_types::Codec
+[Codec64]: mz_persist_types::Codec64
+
+Along with talking to the outside world, STORAGE pilots persist shards around to
+present a "STORAGE collection", which is a durable TVC that handles reshardings
+and version upgrades. A persist shard is exactly a [storage shard] and they can be used
+interchangeably.
+
+[storage shard]: https://github.com/MaterializeInc/materialize/blob/e8c27d00cda02b1f9a50fbfc852d59998b936232/doc/developer/platform/architecture-storage.md#shards
+
+More details available in the [persist design doc].
+
+[persist design doc]: https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/20220330_persist.md
+
+## FAQ: What is persist's throughput?
+
+In general, with proper usage and hardware (and once we finish tuning), persist
+should be able to saturate 75% or more of the available network bandwidth on
+both writes and reads.
+
+Experimentally, the `s3_pg` configuration has sustained 64 MiB/s of goodput for
+10+ hours in an open loop benchmark. This is nowhere near our max, but should
+easily be sufficient for M1. TODO: Add numbers under contention.
+
+    cargo run -p mz-persist-client --bin persist_open_loop_benchmark --blob_uri=... --consensus_uri=...
+
+
+## FAQ: What is persist's latency?
+
+Materialize is not an OLTP database, so our initial tunings are for throughput
+over latency. There are some tricks we can play in the future to get these
+latencies down, but here's a quick idea of where we're starting.
+
+The vertical axis:
+- `mem_mem` uses im-memory implementations of "external durability". These exist
+  for testing but here they're nice because they show the overhead of persist
+  itself.
+- `file_pg` uses files for blob and Postgres for consensus. This is what you
+  might expect in local development or in CI. (These numbers, like the rest, are
+  from a persist-benchmarking ./bin/scratch box. Maybe this one should be from a
```

Comment:
This a question for me or the future reader? :sweat_smile:

Reply:
dunno! future reader maybe? do you have an opinion?

Reply:
No strong opinion, so "future reader" seems fine.